File: joyal670-irohub-loc-fee32c507a5a/app/src/main/java/com/user/student/base/BaseFragment.java
package com.user.student.base;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;
import androidx.viewbinding.ViewBinding;



public abstract class BaseFragment<DB extends ViewBinding> extends Fragment {

    public Context context;
    public FragmentTransactionListener listener;
    public abstract DB getViewBinding();
    DB binding;


    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        this.context = context;
        if (context instanceof FragmentTransactionListener) {
            listener = (FragmentTransactionListener) context;
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
      //  View view = setView(inflater, container, savedInstanceState);
        binding = getViewBinding();
        return binding.getRoot();
    }

  //public abstract View setView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState);

}

File: proteinium/app/src/main/java/com/proteinium/proteiniumdietapp/base/BaseFragment.kt
package com.proteinium.proteiniumdietapp.base

import android.content.Context
import android.content.res.Configuration
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.proteinium.proteiniumdietapp.dialogs.CustomProgressDialog

abstract class BaseFragment : Fragment() {

    var appCompatActivity: AppCompatActivity? = null
    protected lateinit var customProgressDialog: CustomProgressDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.e("oncreate base", "iii")
        appCompatActivity = activity as AppCompatActivity
        customProgressDialog = CustomProgressDialog()
    }

    abstract fun setView(
        inflater: LayoutInflater?,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View?

    abstract fun initData()
    abstract fun setupUI()
    abstract fun setupViewModel()
    abstract fun setupObserver()
    abstract fun onClicks()


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return setView(inflater, container, savedInstanceState)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        setupViewModel()
        setupObserver()
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initData()
        setupUI()
        onClicks()
        adjustFontScale(resources.configuration)
    }

    open fun adjustFontScale(configuration: Configuration) {
        Log.e("font", "adjustFontScale: ${configuration.fontScale}")
        if (configuration.fontScale > 1.1) {
            configuration.fontScale = 1f
            val metrics = resources.displayMetrics
            val wm = context?.getSystemService(AppCompatActivity.WINDOW_SERVICE) as WindowManager
            wm.defaultDisplay.getMetrics(metrics)
            metrics.scaledDensity = configuration.fontScale * metrics.density
            context?.resources?.updateConfiguration(configuration, metrics)
        }
        if (configuration.fontScale < 1.1) {
            configuration.fontScale = 1f
            val metrics = resources.displayMetrics
            val wm = context?.getSystemService(AppCompatActivity.WINDOW_SERVICE) as WindowManager
            wm.defaultDisplay.getMetrics(metrics)
            metrics.scaledDensity = configuration.fontScale * metrics.density
            context?.resources?.updateConfiguration(configuration, metrics)
        }
    }

    fun dismissLoader() {
        customProgressDialog.dialog.dismiss()
    }

    fun showLoader(context: Context? = null) {
        customProgressDialog.show(requireContext())
    }

}